bigearthnet_encoder/encoder.py

```diff
@@ -137,6 +137,21 @@ def write_S1_lmdb(
     )
 
 
+@fc.delegates(_write_lmdb, but=["lmdb_path"])
+def write_S1_S2_lmdb(
+    ben_s1_path: DirectoryPath,
+    ben_s2_path: DirectoryPath,
+    /,
+    lmdb_path: Path = Path("S2_lmdb.db"),
+    **kwargs,
+):
+    patch_paths_s2 = get_s2_patch_directories(ben_s2_path)
+    _write_lmdb(patch_paths_s2,
+                lambda s2_patch_path, **kwargs : tiff_dirs_to_ben_s1_s2_patch(ben_s1_path, s2_patch_path, **kwargs),
```

Comment:
I feel like it is a bit hard to read.
I would prefer to use `partial/fc.partialler` over this long lambda expression.

My personal preference would be to write a _function-builder_ for `tiff_dirs_to_ben_s1_s2_patch`.
So it would be a closure and be initialized only with the S1 path and returns a function that does the same logic as the current version.
Currently, it looks like `tiff_dirs_to_ben_s1_s2_patch` should/could change the _source s1_ dir for every call.

Reply:
Added a closure to generate the tiff_dirs_to_ben_s1_s2_patch with an initial ben_s1_path in the new commit. I do not like the name I went with, so if you would like to rename this function, please feel free to do so.